### PR TITLE
State table

### PR DESCRIPTION
**Pull request by ThatGalNatalie**
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [ ] add/update tests (or don't, for reasons explained below)

## Describe this PR
Enables you to sort by state in the task run table

### CODE REVIEW

File: src/graphql/FlowRun/table-task-runs.gql [hunk 1]
```diff
@@ -9,7 +10,7 @@ query TableTaskRuns(
     id
     name
     task_runs(
-      where: { task: { name: { _ilike: $name } } }
+      where: { task: { name: { _ilike: $name } }, state: { _ilike: $state } }
```
Comment (znicholasbrown) [inline src/graphql/FlowRun/table-task-runs.gql:13]:
```suggestion
      where: { task: { name: { _ilike: $name } }, state: { _ilike: $state } }
```

I don't think we need to do a fuzzy search here, which I think will be slower; we can use `_eq` for state matching instead.

File: src/pages/FlowRun/TaskRunTable-Tile.vue [hunk 2]
```diff
@@ -136,6 +139,35 @@ export default {
...
+        :items="[
+          { header: 'Finished States' },
+          { divider: '...' },
+          'Failed',
+          'Success',
+          'Cancelled',
+          'Finished',
+          'Skipped',
+          'TimedOut',
+          { header: 'Scheduled States - To Re-run Task Run' },
```
Comment (znicholasbrown) [inline src/pages/FlowRun/TaskRunTable-Tile.vue:161]:
```suggestion
          { header: 'Scheduled States - To Re-run Task Run' },
```
I'm not sure this header makes sense for a sort label?

File: src/pages/FlowRun/TaskRunTable-Tile.vue [hunk 3]
```diff
@@ -136,6 +139,35 @@ export default {
...
+          { divider: '...' },
+          'Scheduled',
+          'Resume',
+          { header: 'Pending - to clear the state' },
```
Comment (znicholasbrown) [inline src/pages/FlowRun/TaskRunTable-Tile.vue:165]: I'm not sure this header makes sense for a sort label?

**Review by znicholasbrown — CHANGES_REQUESTED** [review]
Recognizing this is in draft but I let a couple quick comments!

File: src/pages/FlowRun/TaskRunTable-Tile.vue (hunk 2 above)
Comment (ThatGalNatalie) [reply src/pages/FlowRun/TaskRunTable-Tile.vue:161]: oh my bad, I forgot to change that. I copied it from the set state modal. For the sort, I don't think we would need headers?

**Review by znicholasbrown — APPROVED** [review]
This looks great @ThatGalNatalie !

**Pull request merged**